--- AfterStep-2.2.3/Makefile.in~	2006-03-31 23:30:29.000000000 +0200
+++ AfterStep-2.2.3/Makefile.in	2006-11-01 01:45:09.000000000 +0100
@@ -10,6 +10,23 @@
 gtk_app_dirs	= src/tools/ASFileBrowser src/tools/ASWallpaper
 no_gtk_app_dirs	= 
 
+man_subdirs	= libAfterBase libAfterImage libAfterStep libAfterConf @LIBASGTKDIR@ \
+		  src/afterstep \
+		  src/Animate \
+		  src/Arrange \
+		  src/Banner \
+		  src/Ident \
+		  src/Pager \
+		  src/Wharf \
+		  src/WinCommand \
+		  src/WinList2 \
+		  src/WinTabs \
+		  src/ascp  \
+		  src/tools/ASRun \
+		  $(@GTK_APP_DIRS@) \
+		  afterstep \
+		  src/ASDocGen \
+
 subdirs		= libAfterBase libAfterImage libAfterStep libAfterConf @LIBASGTKDIR@ \
 		  src/afterstep \
 		  src/Animate \
@@ -80,7 +97,7 @@
 	@for I in ${libdirs}; do if [ -d $$I ]; then (cd $$I; ${MAKE} install.lib || exit 1); fi; done
 
 install.man: mkdir.man
-	@for I in ${subdirs}; do if [ -d $$I ]; then (cd $$I; ${MAKE} install.man || exit 1); fi; done
+	@for I in ${man_subdirs}; do if [ -d $$I ]; then (cd $$I; ${MAKE} install.man || exit 1); fi; done
   
 install.script: mkdir.bin
 	@for I in ${subdirs}; do if [ -d $$I ]; then (cd $$I; ${MAKE} install.script || exit 1); fi; done
