diff -Nru AfterStep-1.7.111/src/Wharf/Wharf.h AfterStep-1.7.111.new/src/Wharf/Wharf.h
--- AfterStep-1.7.111/src/Wharf/Wharf.h	Mon Mar  8 17:46:40 1999
+++ AfterStep-1.7.111.new/src/Wharf/Wharf.h	Mon Jul 19 08:36:41 1999
@@ -20,8 +20,8 @@
 #define BUTTON_ARRAY_LN 102
 #define FOLDER_ARRAY_LN 10
 
-#define ICON_WIN_WIDTH	48
-#define ICON_WIN_HEIGHT 48
+#define ICON_WIN_WIDTH	55
+#define ICON_WIN_HEIGHT 57
 
 #define ISMAPPED 0
 #define NOTMAPPED 1
