diff -ur AfterStep-2.2.11/libAfterBase/asvector.c AfterStep-2.2.11.new/libAfterBase/asvector.c
--- AfterStep-2.2.11/libAfterBase/asvector.c	2008-12-02 23:35:37.000000000 +0100
+++ AfterStep-2.2.11.new/libAfterBase/asvector.c	2016-11-14 03:08:00.612801550 +0100
@@ -126,7 +126,7 @@
 
 /* finds index of the first element in the vector that is exactly matching specifyed
  * data */
-inline size_t vector_find_data( ASVector *v, void *data )
+size_t vector_find_data( ASVector *v, void *data )
 {
     register int i ;
     /* word copying is usually faster then raw memory copying */
diff -ur AfterStep-2.2.11/libAfterBase/parse.c AfterStep-2.2.11.new/libAfterBase/parse.c
--- AfterStep-2.2.11/libAfterBase/parse.c	2009-02-12 00:32:02.000000000 +0100
+++ AfterStep-2.2.11.new/libAfterBase/parse.c	2016-11-14 03:06:32.054859699 +0100
@@ -1027,7 +1027,7 @@
 
 static char _as_hex_to_char_table[] = "0123456789ABCDEF";
 
-inline int
+int
 unsigned_int2buffer_end (char *buffer, int buffer_size, unsigned int val)
 {
 	int i = buffer_size-1 ; 
--- afterstep-devel-2.2.12/libAfterStep/functions.c.orig	2023-12-21 21:27:43.791252318 +0100
+++ afterstep-devel-2.2.12/libAfterStep/functions.c	2023-12-22 23:16:30.917822883 +0100
@@ -307,7 +307,7 @@ void dup_func_data (FunctionData * dst,
 	}
 }
 
-inline FunctionData *create_named_function (int func, char *name)
+FunctionData *create_named_function (int func, char *name)
 {
 	FunctionData *fdata = safecalloc (1, sizeof (FunctionData));
 
--- afterstep-devel-2.2.12/libAfterStep/mylook.c.orig	2023-12-21 21:27:43.801252264 +0100
+++ afterstep-devel-2.2.12/libAfterStep/mylook.c	2023-12-22 23:19:08.513635780 +0100
@@ -843,7 +843,7 @@ MyStyle *mylook_get_style (MyLook * look
 	return NULL;
 }
 
-inline MyDesktopConfig *mylook_get_desk_config (MyLook * look, long desk)
+MyDesktopConfig *mylook_get_desk_config (MyLook * look, long desk)
 {
 	if (look) {
 		ASHashData hdata = { 0 };
@@ -856,7 +856,7 @@ inline MyDesktopConfig *mylook_get_desk_
 }
 
 
-inline MyBackground *mylook_get_desk_back (MyLook * look, long desk)
+MyBackground *mylook_get_desk_back (MyLook * look, long desk)
 {
 	MyBackground *myback = NULL;
 
@@ -888,7 +888,7 @@ inline MyBackground *mylook_get_desk_bac
 }
 
 
-inline MyBackground *mylook_get_back (MyLook * look, char *name)
+MyBackground *mylook_get_back (MyLook * look, char *name)
 {
 	ASHashData hdata = { 0 };
 	if (look && name)
--- afterstep-devel-2.2.12/src/afterstep/misc.c.orig	2023-12-21 21:27:43.814585525 +0100
+++ afterstep-devel-2.2.12/src/afterstep/misc.c	2023-12-22 23:19:34.623494331 +0100
@@ -32,13 +32,13 @@
 /****************************************************************************/
 /* window management specifics - button ungrabbing convinience functions:   */
 /****************************************************************************/
-inline void ungrab_window_buttons (Window w)
+void ungrab_window_buttons (Window w)
 {
 	LOCAL_DEBUG_OUT ("w = %lX", w);
 	XUngrabButton (dpy, AnyButton, AnyModifier, w);
 }
 
-inline void ungrab_window_keys (Window w)
+void ungrab_window_keys (Window w)
 {
 	XUngrabKey (dpy, AnyKey, AnyModifier, w);
 }
